src/lighteval/models/vllm/vllm_model.py

```diff
@@ -411,3 +420,195 @@ def loglikelihood_rolling():
+            "swap_space": 4,
+            "seed": int(config.seed),
+            "max_num_seqs": int(config.max_num_seqs),
+            "max_num_batched_tokens": int(config.max_num_batched_tokens),
+            "enforce_eager": True,
+        }
+
+        if config.data_parallel_size > 1:
+            self._batch_size = "auto"
+
+        model = AsyncLLM.from_engine_args(AsyncEngineArgs(**self.model_args))
+
+        # If the max_length can't get extracted from the config, it will be inferred from the model
+        if self._max_length is None:
+            self._max_length = model.model_config.max_seq_len_to_capture
+
+        return model
+
+    async def _async_one_item(
+        self,
+        index: int,
+        request: GreedyUntilRequest | LoglikelihoodRequest,
+    ) -> Coroutine[None, list, str]:
+        """Contains the actual logic of the generation."""
+        sampling_params = SamplingParams(**self._config.generation_parameters.to_vllm_dict())
+
+        if isinstance(request, LoglikelihoodRequest):
+            sampling_params.temperature = 0
+            sampling_params.prompt_logprobs = 1
+            sampling_params.max_tokens = 1
+            sampling_params.detokenize = False
+            prompt = request.context + request.choice
+            index = f"logprob_{index}"
+        elif isinstance(request, GreedyUntilRequest):
+            sampling_params.n = request.num_samples
+            sampling_params.max_tokens = self._config.generation_parameters.max_new_tokens or request.generation_size
+            sampling_params.stop = [] if self.use_chat_template else request.stop_sequence
+            sampling_params.logprobs = int(request.use_logits)
+            prompt = request.context
+            index = f"generative_{index}"
+
+        output = await anext(
+            self.model.generate(request_id=str(index), prompt=prompt, sampling_params=sampling_params)
+        )
+        return output
+
+    async def _async_batch(self, requests: list[GreedyUntilRequest | LoglikelihoodRequest]) -> list:
+        # if self.model.engine_core.is_sleeping_async():
+        #    await self.model.engine_core.wake_up_async()
+        processed_requests = [
+            self._async_one_item(index=index, request=request) for index, request in enumerate(requests)
+        ]
+        results = await asyncio.gather(*processed_requests)
+        # await self.model.engine_core.sleep_async()
+        return results
+
+    async def greedy_until(
+        self,
+        requests: list[GreedyUntilRequest],
+        override_bs: Optional[int] = None,
+    ) -> list[GenerativeResponse]:
+        """
+        Generates responses using a greedy decoding strategy until certain ending conditions are met.
+
+        Args:
+            requests (list[Request]): list of requests containing the context and ending conditions.
+            override_bs (int, optional): Override the batch size for generation. Defaults to None.
```

Comment:
I think `batch_size` would be more intuitive here since I initially read `override_bs` as a `bool`

Reply:
It's the parameter we're using across lighteval so I need to keep it homogeneous with the rest of the lib, but agreed it's not a clear name